docs/middleware.md

```diff
@@ -76,6 +76,18 @@ Middleware runs directly after `redirects` and `headers`, before the first files
 
 Middleware uses a [strict runtime](/docs/api-reference/edge-runtime.md) that supports standard Web APIs like `fetch`. This works out of the box using `next start`, as well as on Edge platforms like Vercel, which use [Edge Functions](http://www.vercel.com/edge).
 
+## Custom Server
+
+When using a custom server with middleware, you must specify the hostname and port when instantiating your `NextApp`.
```

Comment:
I wonder if this should be moved to the Custom Server docs, since it only applies to a very small segment of Next.js users who are ejecting for a custom server.